src/main.cpp

```diff
@@ -1435,8 +1435,10 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CTransaction &tx, bool* pfMissingInput
                 // At default rate it would take over a month to fill 1GB
                 if (dFreeCount > GetArg("-limitfreerelay", 15)*10*1000 && !IsFromMe(tx))
                     return error("AcceptToMemoryPool : free transaction rejected by rate limiter");
-                if (fDebug)
-                    LogPrint("mempool", "Rate limit dFreeCount: %g => %g", dFreeCount, dFreeCount+nSize);
+                if (LogInstance().WillLogCategory(BCLog::LogFlags::NOISY))
+                {
+                    LogPrint(BCLog::LogFlags::MEMPOOL, "Rate limit dFreeCount: %g => %g", dFreeCount, dFreeCount+nSize);
```

Comment:
Do we still need the wrapper condition when the call is tagged with a category?

Reply:
Yes. We do. You have to remember that NOISY and VERBOSE are horizontal categories. The original logic here was to require a category turned on (old style) and require debug10. If you collapse this away, then whenever you turn on the mempool category, you will get the noisy entries in that category too.

Reply:
If you do not think these are "noisy" log entries when turning on the mempool category, then we can get rid of the wrapper. There are only five places where we do a mempool LogPrint, and these are two of them.

Reply:
Ah, I see. I imagine that it may be hard to remember which messages need both flags. If I turn on `MEMPOOL`, I may have to go look at the code to remember which messages also need `NOISY`. On the flip side, I may not want to dig through everything from `NOISY` just to read the `MEMPOOL` messages.

Reply:
That is the limitation of this logging system unfortunately. It does not separate "level" from "category". I could rewrite it to have both dimensions, and if I were doing it from scratch that is what I would do, but this is one of the areas that we are trying not to deviate too much from Bitcoin...

Reply:
Removed the wrapper.